extensions/wikia/UserRenameTool/process/ProcessGlobal.class.php

```diff
@@ -0,0 +1,597 @@
+			'phalanx_block_id' => $this->phalanxBlockId,
+			'reason' => $this->reason,
+			'notify_renamed' => $this->notifyUser,
+			'start_time' => $this->startTime,
+		];
+		$task = ( new MultiWikiRename() )->setPriority( PriorityQueue::NAME );
+		$task->call( 'run', $wikiIds, $callParams );
+		$this->currentTaskId = $task->queue();
+	}
+
+	private function antiSpoofTest( $newUserName ) {
+		if ( class_exists( 'SpoofUser' ) ) {
+			$oNewSpoofUser = new \SpoofUser( $newUserName );
+			if ( !$oNewSpoofUser->isLegal() ) {
+				$this->addError( wfMessage( 'userrenametool-error-antispoof-conflict', $newUserName ) );
+			}
+		} else {
+			$this->addError( wfMessage( 'userrenametool-error-antispoof-notinstalled' ) );
+		}
+	}
+
+	private function phalanxTest( $userName ) {
+		$warning = \UserRenameToolHelper::testBlock( $userName );
+		if ( !empty( $warning ) ) {
+			$this->addWarning( $warning );
+		}
+	}
+
+	/**
+	 * Make sure we aren't trying to rename to the same name
+	 *
+	 * @param string $oldUserName
+	 * @param string $newUserName
+	 * @return bool
+	 */
+	private function areUserNamesIdentical( $oldUserName, $newUserName ) {
+		if ( $oldUserName !== $newUserName ) {
+			return false;
+		}
+
+		$this->addError( wfMessage( 'userrenametool-error-same-user' )->inContentLanguage()->text() );
+		return true;
+	}
+
+	private function areUserObjectsValid( $oldUser, $newUser ) {
+		if ( !is_object( $oldUser ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorinvalid',  $this->requestData->oldUsername )
+					->inContentLanguage()
+					->text()
+			);
+			return false;
+		}
+
+		if ( !is_object( $newUser ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorinvalidnew',  $this->requestData->newUsername )
+					->inContentLanguage()
+					->text()
+			);
+			return false;
+		}
+
+		return true;
+	}
+
+	private function addUserDoesNotExistError() {
+		$this->addError(
+			wfMessage( 'userrenametool-errordoesnotexist', $this->requestData->oldUsername )
+				->inContentLanguage()
+				->text()
+		);
+	}
+
+	/**
+	 * @param \User $oldUser
+	 *
+	 * @return bool
+	 */
+	private function checkOldUserPermissions( $oldUser ) {
+		if ( $oldUser->isLocked() ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorlocked', $this->requestData->oldUsername )->inContentLanguage()->text()
+			);
+
+			return false;
+		}
+
+		if ( $oldUser->isAllowed( 'bot' ) ) {
+			$this->addError(
+				wfMessage( 'userrenametool-errorbot', $this->requestData->oldUsername )->inContentLanguage()->text()
+			);
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Find the correct user ID for the old user.  This looks for users who may not have the first letter of their
+	 * user name capitalized.  This became a standard some time ago but users prior to this may exist.
+	 *
+	 * @param string $oldUserName
+	 * @param \User $oldUser
+	 *
+	 * @return int
+	 */
+	private function getOldUserId( $oldUserName, $oldUser ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)